// Proto file representing the EDB Cell class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";

enum CellType {
  CIRCUIT_CELL = 0;
  FOOTPRINT_CELL = 1;
}

service CellService {
  // Creates a cell
  rpc Create (CellCreationMessage) returns (EDBObjMessage) {}

  // Gets the layout of the provided cell
  rpc GetLayout (EDBObjMessage) returns (EDBObjMessage) {}

  // Sets HFSS extents on cell
  rpc SetHfssExtents (CellHfssExtentsMessage) returns (google.protobuf.BoolValue) {}
}

message CellCreationMessage {
  EDBObjMessage database = 1;
  CellType type = 2;
  string name = 3;
}

message ExtentMessage {
  ValueMessage value = 1;
  bool absolute = 2;
}

message CellHfssExtentsMessage {
  EDBObjMessage cell = 1;
  oneof optional_dielectric { ExtentMessage dielectric = 2; }
  oneof optional_airbox_horizontal { ExtentMessage airbox_horizontal = 3; }
  oneof optional_airbox_vertical_positive { ExtentMessage airbox_vertical_positive = 4; }
  oneof optional_airbox_vertical_negative { ExtentMessage airbox_vertical_negative = 5; }
  oneof optional_airbox_truncate_at_ground { google.protobuf.BoolValue airbox_truncate_at_ground = 6; }
}